src/sql-parser/src/parser.rs

```diff
@@ -947,27 +947,43 @@ impl<'a> Parser<'a> {
         } else if let Token::Keyword(kw) = tok {
             match kw {
                 IS => {
+                    let negated = self.parse_keyword(NOT);
                     if self.parse_keyword(NULL) {
```

Comment:
This can be reduced a good bit by using `parse_one_of_keywords` and returning the expected `construct`. You can find some inspiration for this approach in [`parse_create_role`](https://github.com/MaterializeInc/materialize/blob/main/src/sql-parser/src/parser.rs#L2118-L2124).